pandas/core/arrays/integer.py

```diff
@@ -586,14 +587,20 @@ def integer_arithmetic_method(self, other):
 
             op_name = op.__name__
             mask = None
+
+            if getattr(other, 'ndim', 0) > 1:
+                raise NotImplementedError(
+                    "can only perform ops with 1-d structures")
+
             if isinstance(other, (ABCSeries, ABCIndexClass)):
                 other = getattr(other, 'values', other)
 
             if isinstance(other, IntegerArray):
                 other, mask = other._data, other._mask
-            elif getattr(other, 'ndim', 0) > 1:
-                raise NotImplementedError(
-                    "can only perform ops with 1-d structures")
+
+            elif getattr(other, 'ndim', None) == 0:
```

Comment:
This is moved from `dispatch_to_extension_array`.

Reply:
same

Reply:
Had to keep this one in an elif, so that we avoid the `else` block raising a TypeError.